WordPress/src/main/java/org/wordpress/android/ui/posts/mediauploadcompletionprocessors/BlockProcessor.java

```diff
@@ -62,16 +63,18 @@ private Document parseHTML(String blockContent) {
         return document;
     }
 
-    private boolean splitBlock(String block) {
-        Matcher captures = PATTERN_BLOCK_CAPTURES.matcher(block);
+    private boolean splitBlock(String block, Boolean isSelfClosingTag) {
+        Matcher captures = (
+                isSelfClosingTag ? PATTERN_SELF_CLOSING_BLOCK_CAPTURES : PATTERN_BLOCK_CAPTURES
+        ).matcher(block);
 
         boolean capturesFound = captures.find();
 
         if (capturesFound) {
             mBlockName = captures.group(1);
             mJsonAttributes = parseJson(captures.group(2));
-            mBlockContentDocument = parseHTML(captures.group(3));
-            mClosingComment = captures.group(4);
+            mBlockContentDocument = isSelfClosingTag ? null : parseHTML(captures.group(3));
+            mClosingComment = isSelfClosingTag ? null : captures.group(4);
```

Comment:
Blocks with the self-closing tag won't have content so these properties are empty.